lib/dialects/sqlite/query-generator.js

```diff
@@ -79,8 +82,139 @@ const QueryGenerator = {
     return !!value ? 1 : 0;
   },
 
-  addColumnQuery(table, key, dataType) {
+  handleSequelizeMethod(smth, tableName, factory, options, prepend) {
+    if (smth instanceof Utils.Json) {
+      // Parse nested object
+      if (smth.conditions) {
+        const conditions = _.map(this.parseConditionObject(smth.conditions), condition =>
+          `json_extract(${this.quoteIdentifier(_.first(condition.path))}, '$.${_.tail(condition.path).join('.')}') = '${condition.value}'`
+        );
+
+        return conditions.join(' and ');
+      } else if (smth.path) {
+        let str;
+
+        // Allow specifying conditions using the sqlite json1 functions
+        // https://sqlite.org/json1.html
+        if (/(json(?:_.*)*)\(.*\)/i.test(smth.path)) {
```

Comment:
It is derived code from [lib/dialects/postgres/query-generator.js#L150](https://github.com/sequelize/sequelize/blob/master/lib/dialects/postgres/query-generator.js#L150)

And it tests this spec:
```javascript
it('should be able to retrieve a row based on the values of the json document', function() {
  var self = this;

  return this.sequelize.Promise.all([
    this.User.create({ username: 'swen', emergency_contact: { name: 'kate' } }),
    this.User.create({ username: 'anna', emergency_contact: { name: 'joe' } })])
    .then(function() {
      return self.User.find({ where: Sequelize.json(`json_extract(emergency_contact, '$.name')`, 'kate'), attributes: ['username', 'emergency_contact'] });
    })
    .then(function(user) {
      expect(user.emergency_contact.name).to.equal('kate');
    });
});
```
which derived from  [test/integration/dialects/postgres/dao.test.js](https://github.com/sequelize/sequelize/blob/master/test/integration/dialects/postgres/dao.test.js#L221)

I do not understand sequelize as deeply as you, but the original code also seems to check just the query contains raw syntax or not. (means it allows user to use **raw-string query**, or query from ``Literal('...')`` maybe?)  

If this is the case, and If user want to use raw queries, SQL injection should be protected above this **raw-query layer** (before create the query) I think.

If you have any suggestions, please let me know.